src/engine/channelhandle.h

```diff
@@ -143,6 +144,8 @@ class ChannelHandleFactory {
     QHash<ChannelHandle, QString> m_handleToGroup;
 };
 
+typedef std::shared_ptr<ChannelHandleFactory> ChannelHandleFactoryPtr;
```

Comment:
this typedef is misleading.  Please either use ChannelHandleFactoryShrdPtr or don't use the typedef.

Reply:
We use "Pointer"/"Ptr" for many other shared pointers. "ShrdPtr" would be a new variant and is not nice to read.

Reply:
If we want to be consistent we should use "Pointer" instead of introducing another variant.

Reply:
but it's a smart pointer, not a bare one, and that subtlety is lost with the name of the typedef.  Why typedef it at all? it's not ridiculously long as C++ types go.

Reply:
This is how we do it currently in the code base. I agree that this can be optimized, however it is not something we should change in this PR at a single place.  
We may introduce a PR to remove all smart pointer typedefs or whatever we decide.  
This PR is focusing a memory leak, and now the style is consistent.